src/shared/pager.c

```diff
@@ -53,6 +53,11 @@ noreturn static void pager_fallback(void) {
         _exit(EXIT_SUCCESS);
 }
 
+static int stored_stdout = -1;
+static int stored_stderr = -1;
+static bool stdout_redirected = false;
+static bool stderr_redirected = false;
```

Comment:
The pipe needs to be closed. If `stdout_redirected == true` then stdout is the pipe and should be closed (but not necessarily moved back). If `stdout_redirected == false` then stdout is the original stdout and should stay open.
`stdout_redirected` means closing stdout, `stored_stdout >= 0` means moving it back.
Here I'm relying on `dup2(-1, 1) < 0`. Maybe I should change the condition to `if (stdout_redirected && (stored_stdout < 0 || dup2(stored_stdout, STDOUT_FILENO)) < 0)`...